### PR TITLE
Continuing server work for phase 1 integration

### PR DESCRIPTION
**Pull request by dmzimmerman**

### CODE REVIEW

**Comment by kiniry** [comment]
Looks fine so far.

**Comment by dmzimmerman** [comment]
I added a database reset endpoint, /database-reset - POST something there and the database gets reset. Very useful for CDOS testing purposes; probably not something we'll want in production.

File: test/corla-test-credentials.psql [hunk 1]
```diff
@@ -1,82 +1,82 @@
 insert into administrator (id, username, full_name, two_factor_auth_info, last_login_time, type) values 
-(1, 'stateadmin1', 'State Administrator 1', '', null, 'STATE'),
-(2, 'stateadmin2', 'State Administrator 2', '', null, 'STATE'),
-(3, 'stateadmin3', 'State Administrator 3', '', null, 'STATE'),
-(4, 'stateadmin4', 'State Administrator 4', '', null, 'STATE'),
-(5, 'stateadmin5', 'State Administrator 5', '', null, 'STATE'),
-(6, 'stateadmin6', 'State Administrator 6', '', null, 'STATE'),
-(7, 'stateadmin7', 'State Administrator 7', '', null, 'STATE'),
-(8, 'stateadmin8', 'State Administrator 8', '', null, 'STATE');
+(-1, 'stateadmin1', 'State Administrator 1', '', null, 'STATE'),
```
Comment (ranweiler) [inline test/corla-test-credentials.psql]: Were these supposed to all become negative numbers?

File: server/eclipse-project/src/main/java/us/freeandfair/corla/model/RLAAlgorithm.java [hunk 2]
```diff
@@ -144,7 +158,8 @@ public RLAAlgorithm(final CountyDashboard the_dashboard) {
     final Long max_long = 
         CastVoteRecordQueries.countMatching(my_dashboard.cvrUploadTimestamp(), 
                                             my_dashboard.countyID(),
-                                            RecordType.UPLOADED);
+                                            RecordType.UPLOADED) - 1;
```
Comment (ranweiler) [inline server/eclipse-project/src/main/java/us/freeandfair/corla/model/RLAAlgorithm.java]: Why did you make this change? I don't see it documented in the commits.

File: server/eclipse-project/src/main/java/us/freeandfair/corla/endpoint/ResetDatabase.java [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+import us.freeandfair.corla.model.County;
+import us.freeandfair.corla.model.CountyDashboard;
+import us.freeandfair.corla.model.DoSDashboard;
+import us.freeandfair.corla.model.Elector;
+import us.freeandfair.corla.model.UploadedFile;
+import us.freeandfair.corla.persistence.Persistence;
+import us.freeandfair.corla.query.PersistentASMStateQueries;
+
+/**
+ * Reset the database, except for authentication information.
```
Comment (nealmcb) [inline server/eclipse-project/src/main/java/us/freeandfair/corla/endpoint/ResetDatabase.java]: add "and upload artifacts" or the like.

File: test/corla-test-credentials.psql (hunk 1 above)
Comment (dmzimmerman) [inline test/corla-test-credentials.psql]: Yes. It avoids ID conflicts with objects generated automatically by Hibernate.

File: server/eclipse-project/src/main/java/us/freeandfair/corla/model/RLAAlgorithm.java (hunk 2 above)
Comment (dmzimmerman) [inline server/eclipse-project/src/main/java/us/freeandfair/corla/model/RLAAlgorithm.java]: It was wrong in the first place; Joey's code generates a random sequence of the numbers from `min` to `max` inclusive, not from `min` to `max - 1`.

File: server/eclipse-project/src/main/java/us/freeandfair/corla/endpoint/ResetDatabase.java (hunk 3 above)
Comment (dmzimmerman) [inline server/eclipse-project/src/main/java/us/freeandfair/corla/endpoint/ResetDatabase.java]: Sure.

Comment (dmzimmerman) [inline server/eclipse-project/src/main/java/us/freeandfair/corla/endpoint/ResetDatabase.java]: Note that it _does_ remove all _records_ about uploaded artifacts; just not the artifact blobs themselves.

File: test/corla-test-credentials.psql (hunk 1 above)
Comment (ranweiler) [inline test/corla-test-credentials.psql]: :+1:

File: server/eclipse-project/src/main/java/us/freeandfair/corla/model/RLAAlgorithm.java (hunk 2 above)
Comment (ranweiler) [inline server/eclipse-project/src/main/java/us/freeandfair/corla/model/RLAAlgorithm.java]: I suggest adding a comment to this effect.

Comment (dmzimmerman) [inline server/eclipse-project/src/main/java/us/freeandfair/corla/model/RLAAlgorithm.java]: OK, will do.

**Comment by ranweiler** [comment]
@dmzimmerman, just reviewed your pushed changes, which look good. I think this is ready to merge after CI passes.

**Pull request merged**